### ISSUE TITLE
Interested in integration with BBT?

### ISSUE THREAD

**Issue by retorquere** [willsALMANJ/lyz#19]
Hi, I maintain Better BibTeX for Zotero, and I think it'd be interesting to have LyZ and BBT talk to each other. I generate citation keys and do auto-export, I think that's where we overlap. but it may be nice to have the BBT-generated keys available via LyZ.

**Comment by willsALMANJ** [comment]
Being able to talk to BBT would be nice. The main issue for me is finding the time to work on it. I will try to look into what would need to change in LyZ for this when I have time, but you can also look into it if you want.

**Comment by retorquere** [comment]
I'm in the midst of a rewrite for Z5 (10 weeks in and no end in sight) -- I can't currently dive into LyZ. It would be late October at the least before I could. We can park this for now and I can have a look at LyZ then.

**Comment by tmchartrand** [comment]
Just thought I'd chime in on this: I use Lyz with BBT mostly without issue. Seems like there was some integration added a long time ago on the Lyz side, allowing to choose export using BBT and use BBT cite keys when the "generate citekey" is unchecked in Lyz settings. This has for the most part worked fine  for me even through the zotero 5 update, except for a broader Lyz issue with zotero 5 I just discussed in a separate issue.

**Comment by willsALMANJ** [comment]
Yes, nothing in LyZ is BBT specific, but you can choose other export translators, so that is what is going on there.

**Comment by tmchartrand** [comment]
what was pleasantly surprising to me was that the BBT citation keys were still properly found by LyZ following the big rewrite for zotero 5. Nice job on that @retorquere .
The only further integration issues I've found are minor:

1. Journal abbreviations don't seem to work - I'll file an issue once I've investigated further, I'm getting a validation error when abbreviations are turned on.
2. Auto-export support would be nice. BBT could recognize the LyZ-triggered BBT export and allow auto-export of changes following. I imagine this would require significant changes on the LyZ end, calling the full export all at once rather than item by item.

**Comment by retorquere** [comment]
Actually it's more likely LyZ adjusted for the changes in BBT - I think the way to get the keys has changed.

It'd be easy for me to add an API endpoint that would set up an auto-export (and auto-exports are auto-unique on the path exported to, so it's safe to do it more than once), but auto-exports are limited to groups/collections, and there's no easy way to change that. If LyZ is used to grabbing references without having collections in mind, it's going to be difficult to bridge that gap.

**Comment by retorquere** [comment]
What was I thinking - LyZ lives inside zotero, so no api would be needed - just an export with the "Keep updated" option set.
